secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/jws/signer/JwsSigner.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.jws.signer;
+
+import java.util.Map;
+
+import org.forgerock.util.promise.Promise;
+
+/**
+ * Abstraction of the signature methods to implement a code segment to sign messages.
+ * <p>
+ *     As result, the sign task produce a JWS Compact Serialization from JWS, a string composed of three parts encoded in Base64 Url Safe and separated by a dot ( . ), using
+ *     <a href='https://datatracker.ietf.org/doc/html/rfc7159'>JSON-based data structures</a>,
+ *     in accordance with: <a href='https://datatracker.ietf.org/doc/html/rfc7515'>RFC7515 JSON Web Signature (JWS)</a>
+ * </p>
+ * <br/>
+ * <h3>Signer</h3>
+ * <p>
+ * A signer is a functional unity that implements this interface to sign messages. <br/>
+ * A signer implementation is pretended to be configured as IG heap object, where IG will create and initialise it in the heap environment
+ * to be used by the filters and/or the IG scripts. <br/>
+ * <p>
+ *  Reading recommendations
+ *  <ul>
+ *      <li><a href='https://community.forgerock.com/t/identity-gateway-7-0-async-programming-101/29'>Identity Gateway async programming</a></li>
+ *      <li><a href='https://community.forgerock.com/t/identity-gateway-7-1-highway-to-async-programming/28'>Identity Gateway highway to async programming</a></li>
+ *  </ul>
+ * </p>
+ * <h4>Signer heap configuration </h4>
+ * To configure a custom signer implementation as heap object must implement the inner static class named <em>Heaplet</em>, extending {@link org.forgerock.openig.heap.GenericHeaplet}
+ * , that is a generic base class for heaplets with automatically injected fields. <br/>
+ * <ul><li><a href='https://backstage.forgerock.com/docs/ig/7.2/reference/RequiredConfiguration.html#heap-objects'>IG spec reference</a></li></ul>
+ * <br/>
+ * </p>
+ * <p>
```

Comment:
Remove this, it is not specific to the JwsSigner. 

This is useful general advice for IG/SAPI-G extension but doesn't belong here, perhaps we should create a page on the wiki and add it there?